WordPress/gutenberg
# CheckboxControl: Convert component to TypeScript

## What?
Converts the `CheckboxControl` component to TypeScript.

## Why?
Part of the @wordpress/components's TypeScript migration (#35744).

## How?
Refactors the current `CheckboxControl` to TypeScript.

## Testing Instructions
- Run npm run dev and ensure no typing errors
- Fire up the Storybook and confirm the `CheckboxControl` continues to function as expected
- Test the various checkbox controls within the block/site editors in Gutenberg still work
